hs_labels/models.py

```diff
@@ -99,25 +112,37 @@ class UserResourceFlags(models.Model):
 
     user = models.ForeignKey(User, null=False, editable=False,
                              related_name='u2urf',  # unused but must be defined and unique
-                             help_text='user assigning a flag')
+                             help_text='user assigning a flag',
+                             on_delete=models.CASCADE)
+
     resource = models.ForeignKey(BaseResource, null=False, editable=False,
                                  related_name="r2urf",  # unused but must be defined and unique
-                                 help_text='resource to which a flag applies')
+                                 help_text='resource to which a flag applies',
+                                 on_delete=models.CASCADE)
+
+    class Meta:
+        unique_together = (('user', 'resource', 'kind'),)
```

Comment:
Since this a singe set of fields for uniqueness, it can bet set as a single tuple ```('user', 'resource', 'kind')```